src/main/java/mate/academy/bookstore/service/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.bookstore.service.impl;
+
+import lombok.AllArgsConstructor;
+import mate.academy.bookstore.dto.UserRegistrationRequestDto;
+import mate.academy.bookstore.dto.UserResponseDto;
+import mate.academy.bookstore.exception.RegistrationException;
+import mate.academy.bookstore.mapper.UserMapper;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.UserRepository;
+import mate.academy.bookstore.service.LocaleService;
+import mate.academy.bookstore.service.UserService;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserServiceImpl implements UserService {
+    private final UserRepository userRepository;
+    private final UserMapper userMapper;
+    private final LocaleService localeService;
+
+    @Override
+    public UserResponseDto register(UserRegistrationRequestDto requestDto)
+            throws RegistrationException {
+        if (userRepository.findByEmail(requestDto.getEmail()).isPresent()) {
```

Comment:
Can be a bit more efficient and simpler
```suggestion
        if (userRepository.existsByEmail(requestDto.getEmail())) {
```

Reply:
I have read about this method that it is slower and that it is not recommended to use it